### PR TITLE
Docs for BorderBox, Box, Breadcrumb, Button, Counter, Flash, Label

### PR DESCRIPTION
**Pull request by joelhawksley**
This PR adds documentation for the BorderBox, Box, Breadcrumb, Button, Counter, Flash, and Label components, with a few tweaks to our documentation build script along the way.

### CODE REVIEW

**Review by colebemis — COMMENTED** [review]
I ❤️  this docs setup!

I have a few comments:

- I think we should prefer active voice over passive voice for component descriptions (e.g. "Button is used for..." → "Use Button for...")
- I think we should decide how we want to refer to components in the docs (e.g. "breadcrumbs" vs "Breadcrumb" vs "Breadcrumb component" vs "BreadcrumbComponent" vs "Primer::BreadcrumbComponent")
-  As someone new to the project, "kwargs [Hash] Style arguments to be passed to `Primer::Classify`" is not super clear to me. I'm not sure what arguments I'm allowed to pass to the component. Maybe adding a link to the [readme](https://github.com/primer/view_components#built-in-styling-arguments) would help.


If you're not looking for writing style feedback at this point, feel free to ignore me 😄

**Comment by joelhawksley** [comment]
@colebemis thanks for the review!

In general, I copied these component descriptions from the React components. I agree with your language changes, and think we should move the React documentation to the active voice as well.

> I think we should decide how we want to refer to components

I think we should consider whether we want our components to have `Component` in the name of every single object. We're already namespacing all of them as `Primer::...` and having `Component` in the name of every object is getting tiresome.

But generally, I think we should refer to components by as few names as possible. 

> As someone new to the project, "kwargs [Hash] Style arguments to be passed to Primer::Classify" is not super clear

I agree. I've [added](https://github.com/primer/view_components/pull/113/commits/9a1b31338153364045c106274fa9f6fb73d77468) a little helper method to generate a standard link to the style arguments docs.

File: app/components/primer/flash_component.rb [hunk 1]
```diff
@@ -13,7 +14,35 @@ class FlashComponent < Primer::Component
...
+    #       <%= render(Primer::ButtonComponent.new(variant: :small)) { "Take action" } %>
+    #     <% end %>
+    #   <% end %>
+    #
+    # @param full [Boolean] Whether the component should take up the full width of the screen.
+    # @param spacious [Boolean] Whether to add margin to the bottom of the component.
+    # @param dismissible [Boolean] Whether the component can be dismissed with an X button.
+    # @param icon [String] Name of Octicon icon to use.
+    # @param variant [Symbol] <%= one_of(Primer::FlashComponent::VARIANT_MAPPINGS.keys) %>
+    # @param kwargs [Hash] Style arguments to be passed to Primer::Classify
```
Comment (colebemis) [inline app/components/primer/flash_component.rb:45]:
```suggestion
    # @param kwargs [Hash] <%= link_to_style_arguments_docs %>
```

File: app/components/primer/label_component.rb [hunk 2]
```diff
@@ -28,6 +29,22 @@ class LabelComponent < Primer::Component
...
+    #   <%= render(Primer::LabelComponent.new(title: "Label: Label", scheme: :purple)) { "purple" } %>
+    #
+    # @example 40|Variants
+    #   <%= render(Primer::LabelComponent.new(title: "Label: Label")) { "Default" } %>
+    #   <%= render(Primer::LabelComponent.new(title: "Label: Label", variant: :large)) { "Large" } %>
+    #
+    # @param title [String] `title` attribute for the component element.
+    # @param scheme [Symbol] <%= one_of(Primer::LabelComponent::SCHEME_OPTIONS) %>
+    # @param variant [Symbol] <%= one_of(Primer::LabelComponent::VARIANT_OPTIONS) %>
+    # @param kwargs [Hash] Style arguments to be passed to Primer::Classify.
```
Comment (colebemis) [inline app/components/primer/label_component.rb:47]:
```suggestion
    # @param kwargs [Hash] <%= link_to_style_arguments_docs %>
```

**Review by colebemis — APPROVED** [review]

**Pull request merged**